CRM/Utils/Recent.php

```diff
@@ -80,8 +80,18 @@ public static function &get() {
    * Create function used by the API - supplies defaults
    *
    * @param array $params
+   * @param Civi\Api4\Generic\AbstractAction $action
    */
-  public static function create(array $params) {
+  public static function create(array $params, Civi\Api4\Generic\AbstractAction $action) {
+    if ($action->getCheckPermissions()) {
+      $allowed = civicrm_api4($params['entity_type'], 'checkAccess', [
+        'action' => 'get',
+        'values' => ['id' => $params['entity_id']],
+      ], 0);
+      if (empty($allowed['access'])) {
+        return [];
```

Comment:
I thought about throwing an exception here instead of silently doing nothing, but since the original bug report was about a crash caused by an exception, I thought this might be safer?